---
title: "VectorScan - Library characterization"
---

```{r}
# Load relevant libraries 
library(Biostrings)
library('dplyr')
library('corrplot')
library('writexl')
library('readxl')
library('seqinr')
library(ggplot2)
library(gridExtra)
library(ggfortify)
library(pheatmap)
library(Biostrings)
library(tidyverse)
library(igraph)
library(RColorBrewer)
```

LIBRARY CHARACTERIZATION 
----
```{r}
VectorScan <- read_xlsx("VectorScan_Library.xlsx") # Path to the file that contains the library information (e.g., the "Full Table" sheet in 'Table S2.xlsx')
```

In terminal: 
cd [[Location where the raw fastq files from Plate 7 are located]]

Cut off the first part so that only the actual peptide part is there:
for fq in *.fastq.gz; do cutadapt -g TCCAGTCAGGTGTGATGCTCGGGGATCCAGGAATTCGGAGCGGT -o ${fq%%.*}.cut.fastq.gz $fq; done 

With this run, that results in a peptide that is 31 bp long, so make a bowtie index that contains the first 31 bp of each peptide 

```{r}
Bowtie_31bp_fasta <- data.frame(id = VectorScan$id, Seq = substr(VectorScan$'DNA Sequence', 1, 31))
# Create an AAStringSet object
seqs <- AAStringSet(Bowtie_31bp_fasta$Seq)
  # Add names from the sequence header column
names(seqs) <- paste0(Bowtie_31bp_fasta$id)
writeXStringSet(seqs, filepath ="VectorScan_Bowtie_31bp.fasta")
```

Make the bowtie2 index: 
bowtie2-build -f VectorScan_Bowtie_31bp.fasta VectorScan_31bp

Perform the alignment: 
for fq in *.cut.fastq.gz; do bowtie2 -x VectorScan_31bp --very-sensitive -U $fq -S ${fq%%.*}.sam; done

Turn into count files
for i in *.sam; do samtools view -bS $i > ${i%.*}.bam; done
for i in *.bam; do samtools sort $i -o ${i%.*}.sorted.bam; done
for i in *.sorted.bam; do samtools index $i; done
for i in *.sorted.bam; do samtools idxstats $i | cut -f 1,3 | gsed -e'/^\*\t/d' -e'1 i id\t'${i%%.*}| tr "\\t" "," > ${i%%.*}.count.csv; done

paste -d ',' *.count.csv >2025_03_18_VectorScan_Plate07_counts.csv

```{r}
VectorScan_LibraryExpansion <- read.csv('2025_03_18_VectorScan_Plate07_counts.csv', header = TRUE) # Change path to wherever the count file above is located 

VectorScan_LibraryExpansion$Average <- (VectorScan_LibraryExpansion$Rep.1 + VectorScan_LibraryExpansion$Rep.2)/2

num_zeros <- nrow(VectorScan_LibraryExpansion[as.logical(VectorScan_LibraryExpansion$Average <= 1),])
(nrow(VectorScan_LibraryExpansion)-num_zeros)/nrow(VectorScan_LibraryExpansion)*100

original_data <- data.frame(VectorScan_LibraryExpansion$Average)
values_in_range <- original_data[as.logical(original_data[,1] >= 100 & original_data[,1] <= 1000),]
percentage_in_range <- (length(values_in_range) / nrow(original_data)) * 100
percentage_in_range

# Create the histogram of the average number of reads (across both replicates)
ggplot(VectorScan_LibraryExpansion, aes(x = Average)) +
  geom_histogram(bins = 100, fill = "skyblue", color = "black") +
  scale_x_log10() +
  labs(title = "VectorScan input library coverage",
       x = "Count (average of two replicates, log10 Scale)",
       y = "Frequency") +
  theme_minimal()
```
```{r}
VectorScan_GSLinkerNoise <- VectorScan[grep("GGSGGGSGGGSGGGSG", VectorScan$'AA Sequence'), ]
VectorScan_cleaned <- anti_join(VectorScan, VectorScan_GSLinkerNoise, by = 'id')
no_representation <- VectorScan_LibraryExpansion[as.logical(VectorScan_LibraryExpansion$Average == 0),]

VectorScan_cleaned <- anti_join(VectorScan_cleaned, no_representation, by = 'id')

write_xlsx(VectorScan_cleaned, 'VectorScan_cleaned.xlsx')
```